src/libs/searchCountryOptions.js

```diff
@@ -8,15 +8,15 @@ import CONST from '../CONST';
  * @returns {Object[]} An array of countries/states sorted based on the search query
  */
 function searchCountryOptions(searchValue, countriesData) {
-    const trimmedSearchValue = searchValue.toLowerCase().replaceAll(CONST.REGEX.NON_ALPHABETIC_AND_NON_LATIN_CHARS, '');
-    if (trimmedSearchValue.length === 0) {
+    const trimmedSearchValue = _.chain(searchValue).deburr().toLower().value().replaceAll(CONST.REGEX.NON_ALPHABETIC_AND_NON_LATIN_CHARS, '');
+    if (_.isEmpty(trimmedSearchValue)) {
         return [];
     }
 
-    const filteredData = _.filter(countriesData, (country) => country.searchValue.includes(trimmedSearchValue));
+    const filteredData = _.filter(countriesData, (country) => _.includes(_.deburr(country.searchValue), trimmedSearchValue));
```

Comment:
Instead of `_.deburr(country.searchValue)` here, I think we should have fixed the `country.searchValue` here:

https://github.com/Expensify/App/blob/ea944e7db3f2efb2b5b7b68d86f40f0a8895334d/src/components/CountryPicker/CountrySelectorModal.js#L49

What do you think?

Reply:
In my opinion, we can use searchValue: `${countryISO}${countryName}` in CountrySelectorModal,js and searchValue: `${state.stateISO}${state.stateName}` in StateSelectorModal.js and then process the `country.searchValue` in searchCountryOptions.js.

For example: 
App/src/components/CountryPicker/CountrySelectorModal.js
```
...
searchValue: `${countryISO}${countryName}`,
...
```
App/src/components/StatePicker/StateSelectorModal.js
```
...
searchValue: `${state.stateISO}${state.stateName}`,
...
```

src/libs/searchCountryOptions.js
```
function removeDiacriticFromSearchValue(searchValue) {
    return _.chain(searchValue).deburr().toLower().value().replaceAll(CONST.REGEX.NON_ALPHABETIC_AND_NON_LATIN_CHARS, '');
}

function searchCountryOptions(searchValue, countriesData) {
    const trimmedSearchValue = removeDiacriticFromSearchValue(searchValue);
    if (_.isEmpty(trimmedSearchValue)) {
        return [];
    }

    const filteredData = _.filter(countriesData, (country) =>
        _.includes(removeDiacriticFromSearchValue(country.searchValue), trimmedSearchValue),
    );

    // sort by country code
    return _.sortBy(filteredData, (country) => (_.toLower(country.value) === trimmedSearchValue ? -1 : 1));
}
```

What about this?

Reply:
That but instead of:

...
searchValue: `${state.stateISO}${state.stateName}`,
...

do

...
searchValue: removeDiacriticFromSearchValue(`${state.stateISO}${state.stateName}`),
...

Also, I think the name should be a bit more generic since the function is not only replacing diacritics. Something like `prepareSearchValue` or `sanitizeSearchValue`?

Reply:
No problem, please review again. I updated my pull request. @aldo-expensify